include/deal.II/fe/fe_enriched.h

```diff
@@ -644,6 +648,299 @@ class FE_Enriched : public FiniteElement<dim, spacedim>
+   * (hp::FECollection) can be obtained by
+   * <code>fe_collection[index].get_name()</code> and is
+   * show in the table below. Note that all the FE_Enriched elements
+   * are of same size and FE_Nothing<2>() is used as discussed before.
+   *
+   * <table>
+   * <tr>
+   * <th>Active FE index</th>
+   * <th>FiniteElement name</code> </th>
+   * </tr>
+   * <tr>
+   * <td>0</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Nothing<2>()-FE_Nothing<2>()]</code></td>
+   * </tr>
+   * <tr>
+   * <td>1</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Q<2>(1)-FE_Nothing<2>()]</code></td>
+   * </tr>
+   * <tr>
+   * <td>2</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Q<2>(1)-FE_Q<2>(1)]</code></td>
+   * </tr>
+   * <tr>
+   * <td>3</td>
+   * <td><code>FE_Enriched<2>[FE_Q<2>(2)-FE_Nothing<2>()-FE_Q<2>(1)]</code></td>
+   * </tr>
+   * </table>
+   *
+   * The internal data members used by this class need to be available when the
+   * problem is solved. This can be ensured by declaring the object static,
+   * which then remains in the memory until the end of program. An alternative
+   * would be to use it as a data member of the containing class. Since vector
+   * of predicates and enrichment functions may not be available while
+   * constructing the Helper<dim>, a shared_ptr<Helper<dim>> can be used and
+   * constructed when the predicates and enrichment functions are
+   * available.
+   *
+   * Example usage:
+   * @code
+   * FE_Q<dim> fe_base(2);
+   * FE_Q<dim> fe_enriched(1);
+   * std::vector< predicate_function<dim> > predicates;
+   * std::vector< std::shared_ptr<Function<dim>> > enrichments;
+   * hp::FECollection<dim> fe_collection;
+   *
+   * Triangulation<dim>  triangulation;
+   * hp::DoFHandler<dim> dof_handler(triangulation);
+   *
+   * static ColorEnriched::Helper<dim> FE_helper(fe_base,
+   *                                             fe_enriched,
+   *                                             predicates,
+   *                                             enrichments);
+   * FE_helper.set(dof_handler);
+   * fe_collection = FE_helper.get_fe_collection();
+   * @endcode
+   */
+  template<int dim, int spacedim=dim>
+  struct Helper
+  {
+    /**
+     * Constructor for Helper class.
+     *
+     * @param fe_base A base FiniteElement
+     * @param fe_enriched An enriched FiniteElement
+     * @param predicates std::vector of predicates defining the sub-domains.
+     * <code>@p predicates[i]</code> returns true for a cell if it
+     * belongs to a sub-domain with index (i).
+     * @param enrichments std::vector of enrichment functions
+     */
+    Helper(const FE_Q<dim,spacedim> &fe_base,
+           const FE_Q<dim,spacedim> &fe_enriched,
+           const std::vector<predicate_function<dim,spacedim>> &predicates,
+           const std::vector<std::shared_ptr<Function<spacedim>>> &enrichments);
+
+    /**
+     * Prepares @p a hp::DoFHandler object. The active FE indices of
```

Comment:
Remove stray `@p` here. `hp::DoFHandler` will get picked up and rendered as a link by doxygen automatically, by the way.